src/adaptation/icap/icap_log.cc

```diff
@@ -61,10 +61,7 @@ void icapLogLog(AccessLogEntry::Pointer &al)
 {
     if (IcapLogfileStatus == LOG_ENABLE) {
         ACLFilledChecklist checklist(nullptr, al->adapted_request, nullptr);
-        if (al->reply) {
-            checklist.reply = al->reply.getRaw();
-            HTTPMSGLOCK(checklist.reply);
-        }
+        checklist.updateAle(al);
```

Comment:
In places like this, it is possible that the new updateAle() will supply more information to the checklist than the original reply-setting code did. Hopefully, providing access to additional information is an improvement, but it may affect ACL-driven directives in some corner cases (e.g., a server_cert_fingerprint ACL might now match after getting access to ALE::hier.tcpServer that was not available to that ACL before this updateAle() call).